internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -7,52 +7,103 @@ package upgrade
 import (
 	"archive/tar"
 	"archive/zip"
+	"bytes"
 	"compress/gzip"
 	"fmt"
 	"io"
 	"io/fs"
 	"os"
+	"path"
 	"path/filepath"
 	"runtime"
 	"strings"
 
 	"github.com/hashicorp/go-multierror"
 
 	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	v1 "github.com/elastic/elastic-agent/pkg/api/v1"
 	"github.com/elastic/elastic-agent/pkg/core/logger"
 )
 
+type UnpackResult struct {
+	Hash string `json:"hash" yaml:"hash"`
+	// TODO add mapped path of executable
```

Comment:
This TODO still needed?

Reply:
It was a nice-to-have that I abandoned in this first implementation. Will remove the TODO and add some documentation for the struct